src/Bicep.Core/Parsing/BaseParser.cs

```diff
@@ -1230,24 +1233,7 @@ private ObjectTypeSyntax ObjectType(ExpressionFlags flags)
 
         private SyntaxBase ObjectPropertyType(ExpressionFlags flags)
         {
-            List<SyntaxBase> leadingNodes = new();
-
-            // Parse decorators before the declaration.
-            while (this.Check(TokenType.At))
-            {
-                leadingNodes.Add(this.Decorator());
-
-                // All decorators must followed by a newline.
-                leadingNodes.Add(this.WithRecovery(this.NewLine, RecoveryFlags.ConsumeTerminator, TokenType.NewLine));
-
-
-                while (this.Check(TokenType.NewLine))
-                {
-                    // In case there are skipped trivial syntaxes after a decorator, we need to consume
-                    // all the newlines after them.
-                    leadingNodes.Add(this.NewLine());
-                }
-            }
+            var leadingNodes = DecorableSyntaxLeadingNodes().ToImmutableArray();
```

Comment:
I'm sorry, what are the leading nodes for?

Reply:
The leading nodes would be decorators. My thought here was that tuple types can be kind of opaque and can benefit from a `@description()` on individual members. Users could also apply constraint decorators, like:

```bicep
type myTuple = [
  @description('A leading int. Must be positive and two digits')
  @minValue(10)
  @maxValue(99)
  int

  @description('A trailing flag. I should probably explain what this does if I want anyone else to be able to use it.')
  bool
]
```